nistats/first_level_model.py

```diff
@@ -538,26 +538,31 @@ def compute_contrast(self, contrast_def, stat_type=None,
         if len(con_vals) != n_runs:
             warn('One contrast given, assuming it for all %d runs' % n_runs)
             con_vals = con_vals * n_runs
-        if isinstance(output_type, _basestring):
-            if output_type not in ['z_score', 'stat', 'p_value', 'effect_size',
-                                   'effect_variance']:
-                raise ValueError('output_type must be one of "z_score", "stat",'
-                                 ' "p_value","effect_size" or "effect_variance"')
-        else:
-            raise ValueError('output_type must be one of "z_score", "stat",'
-                             ' "p_value","effect_size" or "effect_variance"')
+
+        valid_types = ['z_score', 'stat', 'p_value', 'effect_size',
+                       'effect_variance']
+        if not (isinstance(output_type, _basestring) and
+                output_type in valid_types + ['all']):
+            raise ValueError('output_type must be one of "z_score", "stat", '
+                             '"p_value", "effect_size", "effect_variance" or '
+                             '"all"')
```

Comment:
Use `str.format()` instead.

`ValueError('output_type must be one of {}'.format(valid_types))`